catalog/ui/catalog-ui-search/src/main/webapp/component/singletons/session-timeout.js

```diff
@@ -53,7 +53,9 @@ var sessionTimeoutModel = new (Backbone.Model.extend({
         }
     },
     setPromptTimer: function () {
-        this.promptTimer = setTimeout(this.showPrompt.bind(this), this.get('idleTimeoutDate') - idleNoticeDuration - Date.now());
+        var timeout = this.get('idleTimeoutDate') - idleNoticeDuration - Date.now();
+        timeout = Math.max(0, timeout);
+        this.promptTimer = setTimeout(this.showPrompt.bind(this), timeout);
```

Comment:
❓ Does it make sense to do this same thing for `setLogoutTimer` function below? I'm not entirely sure what the difference is.

Reply:
I thought about it, but since this is really just making sure we don't display <0 I thought I'd keep the change minimal.

Reply:
My impression was that `promptTimer` was the timer that, when it reached 0, would warn the user about impending logout and that `logoutTimer` was the timer that actually triggered the logout.

If it works with this change, though, maybe that's wrong.

Reply:
Per this it looks like I should probably change that as well. https://stackoverflow.com/questions/8430966/is-calling-settimeout-with-a-negative-delay-ok